homeassistant/components/camera/__init__.py

```diff
@@ -991,31 +986,17 @@ async def async_handle_play_stream_service(
     ]
     other_entity_ids = list(set(entity_ids) - set(cast_entity_ids))
 
-    if cast_entity_ids:
-        await hass.services.async_call(
-            DOMAIN_MP,
-            SERVICE_PLAY_MEDIA,
-            {
-                ATTR_ENTITY_ID: cast_entity_ids,
-                **data,
-                ATTR_MEDIA_EXTRA: {
```

Comment:
This is now handled in `async_play_media` of cast media player. It seems a more appropriate place and it now handles both media source as this service.

It allowed me to cleanup this PR too, as the only thing that differs between cast and non-cast is the URL prefix.